test/suites/integration/scheduling_test.go

```diff
@@ -70,353 +72,556 @@ var _ = Describe("Scheduling", Ordered, ContinueOnFailure, func() {
+			// This can result in a case where all 3 pods are healthy, while there are only two created nodes.
+			// In that case, we still expect to eventually have three nodes.
+			env.EventuallyExpectNodeCount("==", 3)
+		})
+		It("should provision a node using a NodePool with higher priority", func() {
+			nodePoolLowPri := test.NodePool(corev1beta1.NodePool{
+				Spec: corev1beta1.NodePoolSpec{
+					Weight: ptr.Int32(10),
+					Template: corev1beta1.NodeClaimTemplate{
+						Spec: corev1beta1.NodeClaimSpec{
+							NodeClassRef: &corev1beta1.NodeClassReference{
+								Name: nodeClass.Name,
 							},
-							{
-								Key:      v1.LabelInstanceTypeStable,
-								Operator: v1.NodeSelectorOpIn,
-								Values:   []string{"t3.nano"},
+							Requirements: []v1.NodeSelectorRequirement{
+								{
+									Key:      v1.LabelOSStable,
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{string(v1.Linux)},
+								},
+								{
+									Key:      v1.LabelInstanceTypeStable,
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"t3.nano"},
+								},
 							},
 						},
 					},
 				},
-			},
-		})
-		nodePoolHighPri := test.NodePool(corev1beta1.NodePool{
-			Spec: corev1beta1.NodePoolSpec{
-				Weight: ptr.Int32(100),
-				Template: corev1beta1.NodeClaimTemplate{
-					Spec: corev1beta1.NodeClaimSpec{
-						NodeClassRef: &corev1beta1.NodeClassReference{
-							Name: nodeClass.Name,
-						},
-						Requirements: []v1.NodeSelectorRequirement{
-							{
-								Key:      v1.LabelOSStable,
-								Operator: v1.NodeSelectorOpIn,
-								Values:   []string{string(v1.Linux)},
+			})
+			nodePoolHighPri := test.NodePool(corev1beta1.NodePool{
+				Spec: corev1beta1.NodePoolSpec{
+					Weight: ptr.Int32(100),
+					Template: corev1beta1.NodeClaimTemplate{
+						Spec: corev1beta1.NodeClaimSpec{
+							NodeClassRef: &corev1beta1.NodeClassReference{
+								Name: nodeClass.Name,
 							},
-							{
-								Key:      v1.LabelInstanceTypeStable,
-								Operator: v1.NodeSelectorOpIn,
-								Values:   []string{"c5.large"},
+							Requirements: []v1.NodeSelectorRequirement{
+								{
+									Key:      v1.LabelOSStable,
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{string(v1.Linux)},
+								},
+								{
+									Key:      v1.LabelInstanceTypeStable,
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"c5.large"},
+								},
 							},
 						},
 					},
 				},
+			})
+			pod := test.Pod()
+			env.ExpectCreated(pod, nodeClass, nodePoolLowPri, nodePoolHighPri)
+			env.EventuallyExpectHealthy(pod)
+			env.ExpectCreatedNodeCount("==", 1)
+			Expect(ptr.StringValue(env.GetInstance(pod.Spec.NodeName).InstanceType)).To(Equal("c5.large"))
+			Expect(env.GetNode(pod.Spec.NodeName).Labels[corev1beta1.NodePoolLabelKey]).To(Equal(nodePoolHighPri.Name))
+		})
+
+		DescribeTable(
```

Comment:
This is 6 tests. Maybe they're all necessary (I hadn't looked into them that deeply) but is there a way to cut it down to a smaller set of tests? If you think they are all necessary, I'm fine taking these changes in.